prow/plugins/lgtm/lgtm.go

```diff
@@ -263,35 +266,68 @@ func handle(wantLGTM bool, config *plugins.Configuration, ownersClient repoowner
 	// Only add the label if it doesn't have it, and vice versa.
 	labels, err := gc.GetIssueLabels(org, repoName, number)
 	if err != nil {
-		log.WithError(err).Errorf("Failed to get the labels on %s/%s#%d.", org, repoName, number)
+		log.WithError(err).Errorf("Failed to get issue labels.")
 	}
 	hasLGTM := github.HasLabel(LGTMLabel, labels)
 
 	// remove the label if necessary, we're done after this
 	if hasLGTM && !wantLGTM {
 		log.Info("Removing LGTM label.")
-		return gc.RemoveLabel(org, repoName, number, LGTMLabel)
-	}
-
-	// add the label if necessary, we're done after this
-	if !hasLGTM && wantLGTM {
+		if err := gc.RemoveLabel(org, repoName, number, LGTMLabel); err != nil {
+			return err
+		}
+		botname, err := gc.BotName()
+		if err != nil {
+			return err
+		}
+		comments, err := gc.ListIssueComments(org, repoName, number)
+		if err != nil {
+			log.WithError(err).Errorf("Failed to get issue comments.")
+		}
+		for _, comment := range comments {
+			m := addLGTMLabelNotificationRe.FindStringSubmatch(comment.Body)
+			if comment.User.Login == botname && m != nil {
+				log.Info("Removing tree-hash comment.")
+				if err := gc.DeleteComment(org, repoName, comment.ID); err != nil {
+					log.WithError(err).Errorf("Failed to delete comment.")
+				}
+			}
+		}
+	} else if !hasLGTM && wantLGTM {
 		log.Info("Adding LGTM label.")
 		if err := gc.AddLabel(org, repoName, number, LGTMLabel); err != nil {
 			return err
 		}
-		// Delete the LGTM removed notification after the LGTM label is added.
+		opts := optionsForRepo(config, rc.repo.Owner.Login, rc.repo.Name)
+		if opts.StoreTreeHash {
+			pr, err := gc.GetPullRequest(org, repoName, number)
+			if err != nil {
+				log.WithError(err).Errorf("Failed to get pull request.")
+			}
+			SHA := *pr.MergeSHA
```

Comment:
`MergeSHA` is a pointer so we need a `nil` check before accessing it.
In particular, this field is lazily calculated and may not always be specified.